packages/core/src/services/evaluations/create.ts

```diff
@@ -194,3 +225,68 @@ async function findProvider(workspace: Workspace, db: Database) {
 
   return providers.find((p) => p.token === env.DEFAULT_PROVIDER_API_KEY)
 }
+
+export async function createAdvancedEvaluation(
+  {
+    workspace,
+    configuration,
+    metadata,
+    ...props
+  }: {
+    workspace: Workspace
+    user: User
+    name: string
+    description: string
+    configuration: EvaluationResultConfiguration
+    metadata: { prompt: string; templateId?: number }
+    projectId?: number
+    documentUuid?: string
+  },
+  db = database,
+): PromisedResult<EvaluationDto> {
+  const validConfig = validateConfiguration(configuration)
+  if (validConfig.error) return validConfig
+
+  const provider = await findProvider(workspace, db)
+  if (!provider) {
+    return Result.error(
+      new NotFoundError(
+        'In order to create an evaluation you need to first create a provider API key from OpenAI or Anthropic',
+      ),
+    )
+  }
+
+  const promptWithProvider = provider
+    ? `---
+provider: ${provider.name}
+model: ${findFirstModelForProvider(provider.provider)}
+---
+${metadata.prompt}
+`.trim()
+    : metadata.prompt
+
+  const resultConfiguration = (
+    configuration.type === EvaluationResultableType.Number
+      ? {
+          minValue: configuration.detail!.range.from,
+          maxValue: configuration.detail!.range.to,
+        }
+      : {}
+  ) as IEvaluationConfiguration
+
+  return createEvaluation(
+    {
+      workspace,
+      ...props,
+      metadataType: EvaluationMetadataType.LlmAsJudgeAdvanced,
+      metadata: {
+        prompt: promptWithProvider,
+        configuration,
+        templateId: metadata.templateId ?? null,
+      } as Omit<EvaluationMetadataLlmAsJudgeAdvanced, 'id'>,
+      resultType: configuration.type,
+      resultConfiguration,
+    },
+    db,
+  )
+}
```

Comment:
there's no way you are merging this without adding tests for this service 😅

Reply:
There are tests for this service! They check that everything is still working as expected.